### PR TITLE
Create example for running history matching in python using analysis module

### PR DESCRIPTION
**Pull request by ManInFez**
**Issue**
Resolves #my_issue

**Approach**
_Short description of the approach_

## Pre review checklist

- [ ] Added appropriate labels

Adding labels helps the maintainers when writing release notes, see sections and the
corresponding labels here: https://github.com/equinor/ert/blob/main/.github/release.yml

### CODE REVIEW

File: docs/rst/manual/advanced/Polynormial.ipynb [hunk 1]
```diff
@@ -0,0 +1,292 @@
+{
```
Comment (berland) [inline docs/rst/manual/advanced/Polynormial.ipynb:1]: typo in filename, Polynormial -> Polynomial

File: docs/rst/manual/advanced/Oscilator.ipynb [hunk 2]
```diff
@@ -0,0 +1,345 @@
+{
```
Comment (berland) [inline docs/rst/manual/advanced/Oscilator.ipynb:1]: typo in filename, Oscilator -> Oscillator

**Comment by dafeda** [comment]
Reviewing jupyter notebooks is difficult without converting them to `.py` files.
I've used post save-hooks in the past to automatically save `.py` and `.html` versions of notebooks as described here:

https://www.svds.com/jupyter-notebook-best-practices-for-data-science/

Do you know of better methods?

File: ert/analysis/ies/__init__.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
+from typing import Optional, Union, Any
+import numpy as np
+import res
+
+# pylint: disable=import-error
+from res._lib.ies import (  # type: ignore
```
[the remark this answers is not in the document]
Comment (ManInFez) [reply ert/analysis/ies/__init__.py:6]: yes, but I want to make functions available in the module like this `ert.analysis.ies.updateA`  and thus hide `res._lib` for the users

File: ert/analysis/ies/__init__.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    make_D,
+    init_update,
+    ModuleData,
+    Config,
+    inversion_type,
+)
+
+
+def make_X(  # pylint: disable=too-many-arguments
+    S: np.typing.NDArray[np.double],
```
[the remark this answers is not in the document]
Comment (ManInFez) [reply ert/analysis/ies/__init__.py:17]: Didnt work for me, but I probably did something wrong

File: docs/rst/manual/advanced/api.rst [hunk 5]
```diff
@@ -776,3 +776,45 @@ ERT script function                                                        Purpo
...
+
+General overview
+~~~~~~~~~~~~~~~~
+
+ERT supports an API for performing history matching based on data represented by numpy arrays.
+
+
+
+
+*History matching example*
```
[the remark this answers is not in the document]
Comment (ManInFez) [reply docs/rst/manual/advanced/api.rst:792]: I wanted to include the notebooks, but I was unable to build docs locally so it was a bit hard to test.

**Comment by ManInFez** [comment]
@dafeda I am not so sure to include the hm_analysis example from storage in the poly_example. Any thoughts where this could be included

File: docs/rst/manual/advanced/api.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (dafeda) [reply docs/rst/manual/advanced/api.rst:792]: `sphinx-build -n -v -E -W ./docs/rst/manual ./tmp/ert_docs` no work?

File: ert/analysis/ies/__init__.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ManInFez) [reply ert/analysis/ies/__init__.py:17]: sure

**Review by dafeda — APPROVED** [review]
LGTM 👍

**Pull request merged**